sys/include/net/ng_ipv6/netif.h

```diff
@@ -0,0 +1,251 @@
+ * @return  -ENOMEM, if there is no space left to store @p addr.
+ */
+int ng_ipv6_netif_add_addr(kernel_pid_t pid, const ng_ipv6_addr_t *addr,
+                           bool anycast);
+
+/**
+ * @brief   Remove an address from the interface.
+ *
+ * @param[in] pid       The PID to the interface.
+ * @param[in] addr      An address you want to remove from interface.
+ */
+void ng_ipv6_netif_remove_addr(kernel_pid_t pid, ng_ipv6_addr_t *addr);
+
+/**
+ * @brief   Removes all addresses from the interface.
+ *
+ * @param[in] pid       The PID to the interface.
+ */
+void ng_ipv6_netif_reset_addr(kernel_pid_t pid);
+
+/**
+ * @brief   Searches for an address on all interfaces.
+ *
+ * @param[out] out  The reference to the address on the interface.
+ * @param[in] addr  The address you want to search for.
+ *
+ * @return  The PID to the interface the address is registered to.
+ * @return  KERNEL_PID_UNDEF, if the address can not be found on any interface.
+ */
+kernel_pid_t ng_ipv6_netif_find_addr(ng_ipv6_addr_t **out, const ng_ipv6_addr_t *addr);
+
+/**
+ * @brief   Searches for an address on an interface.
+ *
+ * @param[in] pid   The PID to the interface.
+ * @param[in] addr  The address you want to search for.
+ *
+ * @return  The reference to the address on the interface.
+ * @return  NULL, if the address can not be found on the interface.
+ * @return  NULL, if @p pid is no interface.
+ */
+ng_ipv6_addr_t *ng_ipv6_netif_find_addr_local(kernel_pid_t pid, const ng_ipv6_addr_t *addr);
+
+/**
+ * @brief   Searches for the first address matching a prefix best on all
+ *          interfaces.
+ *
+ * @param[out] out      The reference to the found address on the interface.
+ * @param[in] prefix    The prefix you want to search for.
+ *
+ * @return  The PID to the interface the address is registered to.
+ * @return  KERNEL_PID_UNDEF, if no matching address can not be found on any
+ *          interface.
+ */
+kernel_pid_t ng_ipv6_netif_match_prefix(ng_ipv6_addr_t **out, const ng_ipv6_addr_t *prefix);
+
+/**
+ * @brief   Searches for the first address matching a prefix best on an
+ *          interfaces.
+ *
+ * @param[in] pid       The PID to the interface.
+ * @param[in] prefix    The prefix you want to search for.
+ *
+ * @return  The reference to the found address on the interface.
+ * @return  NULL, if no matching address can be found on the interface.
+ * @return  NULL, if @p pid is no interface.
+ */
+ng_ipv6_addr_t *ng_ipv6_netif_match_prefix_local(kernel_pid_t pid,
+        const ng_ipv6_addr_t *prefix);
+
+/**
+ * @brief   Searches for the best address on an interface usable as a
+ *          source address for a given destination address.
+ *
+ * @param[in] pid   The PID to the interface.
+ * @param[in] dest  The destination address you want to find a destination
+ *                  address for.
+ *
+ * @return  The reference to the found address on the interface.
+ * @return  NULL, if no matching address can be found on the interface.
```

Comment:
Though it might be obvious, I think the documentation should state that NULL will be returned if dest is not a unicast address.